cachito/workers/config.py

```diff
@@ -2,15 +2,53 @@
 import logging
 import os
 import tempfile
-from typing import Dict, List, Optional
+from typing import Dict, List, Optional, cast
 
 import celery
 import kombu
+from celery.signals import worker_process_init
+from opentelemetry import trace
+from opentelemetry.exporter.jaeger.thrift import JaegerExporter
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+from opentelemetry.instrumentation.celery import CeleryInstrumentor
+from opentelemetry.instrumentation.requests import RequestsInstrumentor
+from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
 
 from cachito.errors import ConfigError
 
 ARCHIVES_VOLUME = os.path.join(tempfile.gettempdir(), "cachito-archives")
 
+RequestsInstrumentor().instrument()
+
+
+@worker_process_init.connect(weak=False)
+def init_celery_tracing(*args, **kwargs):
```

Comment:
We generally do these sort of celery hooks in https://github.com/containerbuildsystem/cachito/blob/13876fe66c0a53562bb4bbe8af22bb22290ec4f7/cachito/workers/tasks/celery.py#L24-L28

Maybe that would also solve

>     # Some of the following has already been executed due to the manner in which
>     # the tasks config is included....

?